spec/controllers/sessions_controller_spec.rb

```diff
@@ -0,0 +1,147 @@
+describe SessionsController do
+
+  context "request for login page" do
+    it "has a successful response" do
+      get :new
+
+      expect(response).to have_http_status(:success)
+      expect(response.content_type).to eq("text/html")
+    end
+
+    it "renders the new session template" do
+      get :new
+
+      expect(response).to render_template(:new)
+    end
+  end
+
+  context "login successful" do
+    before(:each) do
+      User.create(username: "chosen", email: "thechosenone@example.com", password: "guitarman")
+    end
+
+    it "has a successful response" do
+      post :create, session: { email: "thechosenone@example.com", password: "guitarman" }
+
+      expect(response).to have_http_status(:redirect)
+      expect(response.content_type).to eq("text/html")
+    end
+
+    it "redirects to the item index page" do
+      post :create, session: { email: "thechosenone@example.com", password: "guitarman" }
+
+      expect(response).to redirect_to items_path
+    end
+
+    it "creates a session for the user" do
+      post :create, session: { email: "thechosenone@example.com", password: "guitarman" }
+
+      expect(session[:user_id]).to eq(User.last.id)
```

Comment:
If you need the user to be created before calling `user`, you can use a `let!`, but some consider that poor form. The best way might be to just have a method:

```ruby
user = create_user
# your test
```